src/mixins/java/org/spongepowered/common/mixin/core/world/level/ExplosionMixin.java

```diff
@@ -118,158 +113,107 @@ public abstract class ExplosionMixin implements ExplosionBridge {
-
-        for (int k2 = 0; k2 < list.size(); ++k2) {
-            final Entity entity = list.get(k2);
-            if (!entity.ignoreExplosion()) {
-                final double d12 = (double)(Mth.sqrt(entity.distanceToSqr(vec3d)) / f3);
-                if (d12 <= 1.0D) {
-                    double d5 = entity.getX() - this.x;
-                    double d7 = entity.getEyeY() - this.y;
-                    double d9 = entity.getZ() - this.z;
-                    final double d13 = (double)Mth.sqrt(d5 * d5 + d7 * d7 + d9 * d9);
-                    if (d13 != 0.0D) {
-                        d5 = d5 / d13;
-                        d7 = d7 / d13;
-                        d9 = d9 / d13;
-                        final double d14 = (double) net.minecraft.world.level.Explosion.getSeenPercent(vec3d, entity);
-                        final double d10 = (1.0D - d12) * d14;
-                        entity.hurt(this.shadow$getDamageSource(), (float)((int)((d10 * d10 + d10) / 2.0D * 7.0D * (double)f3 + 1.0D)));
-                        double d11 = d10;
-                        if (entity instanceof LivingEntity) {
-                            d11 = ProtectionEnchantment.getExplosionKnockbackAfterDampener((LivingEntity)entity, d10);
-                        }
-
-                        // Sponge Start - Honor our knockback value from event
-                        entity.setDeltaMovement(entity.getDeltaMovement().add(d5 * d11 * this.impl$knockback, d7 * d11 * this.impl$knockback, d9 * d11 * this.impl$knockback));
-                        if (entity instanceof Player) {
-                            final Player playerentity = (Player)entity;
-                            if (!playerentity.isSpectator() && (!playerentity.isCreative() || !playerentity.abilities.flying)) {
-                                this.hitPlayers.put(playerentity, new Vec3(d5 * d10 * this.impl$knockback,
-                                        d7 * d10 * this.impl$knockback, d9 * d10 * this.impl$knockback));
-                            }
-                        }
-                        // Sponge End
-                    }
-                }
-            }
-        }
+    }
+
+    @Redirect(
+            method = "explode",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/world/phys/Vec3;add(DDD)Lnet/minecraft/world/phys/Vec3;",
+                    ordinal = 0
+            )
+    )
+    public Vec3 impl$explode_applyKnockbackModifier(Vec3 instance, double param0, double param1, double param2) {
+        return instance.add(param0 * this.impl$knockback, param1 * this.impl$knockback, param2 * this.impl$knockback);
+    }
+
+    @Redirect(
+            method = "explode",
+            at = @At(value = "NEW", target = "net/minecraft/world/phys/Vec3", ordinal = 1)
+    )
+    public Vec3 impl$explode_recordKnockbackModifier(double param0, double param1, double param2) {
```

Comment:
final parameters and private method declaration. Give the parameters names more appropriate than just `param0` etc.